### PR TITLE
Make 0.12.1.x minable again

### PR DESCRIPTION
**Pull request by UdjinM6**
Turned out that 0.12.1.x is _technically_ not fully minable by pools now since pools can't get a single idea of what kind of superblock payment they have to include to pass block validation. And at high diff like on mainnet it means it's not minable at all - no cpu out there can solve a single block in reasonable time to move chain forward.

Changes are for `getblocktemplate`, `CreateNewBlock`, `FillBlockPayee`, `CreateSuperblock`:
- Renamed global `FillBlockPayee` to `FillBlockPayments`;
- `FillBlockPayments`, `FillBlockPayee` and `CreateSuperblock` now return additional info back via arguments;
- Add support for superblocks in `getblocktemplate`;
- `CreateNewBlock` adjusted: regular coinbase is now completely calculated here and only adjusted by `FillBlockPayee` and `CreateSuperblock` (this also fixes miner reward for superblocks - it was missing);
- Refactor the way masternode payments are passed around, change `getblocktemplate` format for them too.

NOTE: changing `getblocktemplate` format for masternodes actually makes 0.12.1.x non-minable by current versions of mining pools but imo that's fine and the right way to go - we must make sure that pools would not just replace dashd 0.12.0.x with dashd 0.12.1.x but would actually update their software too if they run 0.12.1.x. Leaving it backwards compatible is risky imo - we could fork badly on first superblock.

Side note: `SPORK_9_MASTERNODE_SUPERBLOCK_ENFORCEMENT` should really be renamed to `SPORK_9_SUPERBLOCKS_ENABLED`... but not in this PR :)

### CODE REVIEW

**Comment by tgflynn** [comment]
utACK

File: src/rpcmining.cpp [hunk 1]
```diff
@@ -381,10 +381,26 @@ UniValue getblocktemplate(const UniValue& params, bool fHelp)
...
-            "  \"masternode_payments\" : true|false,         (boolean) true, if masternode payments are enabled\n"
-            "  \"enforce_masternode_payments\" : true|false  (boolean) true, if masternode payments are enforced\n"
+            "  \"masternode\" : [                  (array) required masternode payee that must be included in the next block\n"
+            "      {\n"
+            "         \"payee\" : \"xxxx\",          (string) payee address\n"
+            "         \"script\" : \"xxxx\",         (string) payee scriptPubKey\n"
+            "         \"amount\": n                (numeric) required amount to pay\n"
+            "      }\n"
+            "      ,...\n"
+            "  ],\n"
```
Comment (thelazier) [inline src/rpcmining.cpp]: compiled locally and testing, found that getblocktemplate result is
```  "masternode": {
    "payee": "yUuAsYCnG5XrjgsGvRwcDqPhgLUnzNfe8L",
    "script": "76a9145e21cfe3b3ce2e6027bff8c73db872a89561871188ac",
    "amount": 1250000000
  },
```
Are we having more than one masternode payment object in a block ? (for `,...`)

**Comment by thelazier** [comment]
May I see the expected example of getblocktemplate ? Is it correct by the code or Is it expected to be as in `dash-cli help getblocktemplate` ?

File: src/rpcmining.cpp (hunk 1 above)
Comment (UdjinM6) [inline src/rpcmining.cpp]: Nicely spotted! 👍 I was thinking of having more than one and was playing with that a bit too to see how that might fit but then removed that from code, forgot to cleanup description :) Will fix.

**Comment by UdjinM6** [comment]
Fixed getblocktemplate description for `masternode`, should match the actual code now.
Tnx @thelazier ! 👍

**Pull request merged**